primitives/src/targeting/eval.rs

```diff
@@ -401,6 +518,81 @@ fn eval(input: &Input, output: &mut Output, rule: &Rule) -> Result<Option<Value>
 
             Some(Value::Bool(a && b))
         }
+        Function::Or(first_rule, second_rule) => {
+            let a = eval(input, output, first_rule)?
+                .ok_or(Error::TypeError)?
+                .try_bool()?;
+            let b = eval(input, output, second_rule)?
+                .ok_or(Error::TypeError)?
+                .try_bool()?;
+
+            Some(Value::Bool(a || b))
+        }
+        Function::Xor(first_rule, second_rule) => {
+            let a = eval(input, output, first_rule)?
+                .ok_or(Error::TypeError)?
+                .try_bool()?;
+            let b = eval(input, output, second_rule)?
+                .ok_or(Error::TypeError)?
+                .try_bool()?;
+
+            Some(Value::Bool((a && !b) || (!a && b)))
+        }
+        Function::Not(first_rule) => {
+            let a = eval(input, output, first_rule)?
+                .ok_or(Error::TypeError)?
+                .try_bool()?;
+
+            Some(Value::Bool(!a))
+        }
+        Function::Lt(first_rule, second_rule) => {
```

Comment:
these functions should work the same way as Math functions.
We've discussed it, this comment is just a reminder.